cli/store_cmds.go

```diff
@@ -244,6 +244,57 @@ func Import(buildDataFS vfs.FileSystem, stor interface{}, opt ImportOpt) error {
 		hasIndexableData bool
 	)
 
+	importGraphData := func(graphFile string, sourceUnit *unit.SourceUnit) error {
+		var data graph.Output
+		if err := readJSONFileFS(buildDataFS, graphFile, &data); err != nil {
+			if err == errEmptyJSONFile {
+				log.Printf("Warning: the JSON file is empty for unit %s %s.", sourceUnit.Type, sourceUnit.Name)
+				return nil
+			}
+			if os.IsNotExist(err) {
+				log.Printf("Warning: no build data for unit %s %s.", sourceUnit.Type, sourceUnit.Name)
+				return nil
+			}
+			return fmt.Errorf("error reading JSON file %s for unit %s %s: %s", graphFile, sourceUnit.Type, sourceUnit.Name, err)
+		}
+		if opt.DryRun || GlobalOpt.Verbose {
+			log.Printf("# Importing graph data (%d defs, %d refs, %d docs, %d anns) for unit %s %s", len(data.Defs), len(data.Refs), len(data.Docs), len(data.Anns), sourceUnit.Type, sourceUnit.Name)
+			if opt.DryRun {
+				return nil
+			}
+		}
+
+		// HACK: Transfer docs to [def].Docs.
+		docsByPath := make(map[string]*graph.Doc, len(data.Docs))
+		for _, doc := range data.Docs {
+			docsByPath[doc.Path] = doc
+		}
+		for _, def := range data.Defs {
+			if doc, present := docsByPath[def.Path]; present {
+				def.Docs = append(def.Docs, &graph.DefDoc{Format: doc.Format, Data: doc.Data})
+			}
+		}
+
+		switch imp := stor.(type) {
+		case store.RepoImporter:
+			if err := imp.Import(opt.CommitID, sourceUnit, data); err != nil {
+				return fmt.Errorf("error running store.RepoImporter.Import: %s", err)
+			}
+		case store.MultiRepoImporter:
+			if err := imp.Import(opt.Repo, opt.CommitID, sourceUnit, data); err != nil {
+				return fmt.Errorf("error running store.MultiRepoImporter.Import: %s", err)
+			}
+		default:
+			return fmt.Errorf("store (type %T) does not implement importing", stor)
+		}
+
+		mu.Lock()
```

Comment:
Why does this mutex exist at all? The assignment is atomic and the mutex is not consulted when the value is read. Shouldn't we just delete the mutex and add a comment to the var saying where it could be modified/read concurrently?